proxy/handlers.go

```diff
@@ -147,3 +148,47 @@ func (p *Proxy) Impersonate(w http.ResponseWriter, r *http.Request) {
 
 	http.Redirect(w, r, dashboardURL, http.StatusFound)
 }
+
+// Verify checks a user's credentials for an arbitrary host. If the user is
+// properly authenticated and is authorized to access the supplied host,
+// a 200 http status code is returned. If the user is not authenticated, they
+// will be redirected to the authenticate service to sign in with their identity
+// provider. If the user is unauthorized, they will be given a 403 http status.
+func (p *Proxy) Verify(w http.ResponseWriter, r *http.Request) {
```

Comment:
It seems odd to run the proxy just to bounce auth checks off of.  It feels like something Authorize should do...but that's obviously a whole extra set of problems.

In a typical config where all pomerium services share the policy, the proxy service would function if you simply pointed your `from` addresses at it, correct?  Perhaps we should disable proxying in this mode?

Reply:
This is where #291 would potential come in. We could skip setting up the reverse proxy if no `to` destination was defined in the policy (but still allow all the `.pomerium/` routes for troubleshooting and verification.

Plus, this would allow for a mixed model of some routes using pomerium directly for proxying, while other not, if that was a desired configuration setup. 

What do you think about that?

Reply:
Hmm.  In this mode, would any policy be defined at all on the proxy?  It strikes me that it just serves as an integration point to Authorize and Authenticate.  The proxy wouldn't care about the `from` at all... except maybe an unprotected endpoint?

From a deployment perspective where you might be sharing the config file, it would probably be nice to be able to set this mode with a single flag.  That also makes the decision explicit.  Thoughts?

Reply:
I've added a flag that sits on the main option  (vs policy) branch. I went back and forth as to whether this should be under policy or config, but it feels easier to implement in `config.

Reply:
I think main config makes sense.